kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/ALBookingRMAdapter.java

```diff
@@ -0,0 +1,534 @@
+                viewHolder.productImageOverlay.setVisibility(GONE);
+                viewHolder.productPrice.setBackground(context.getResources().getDrawable(R.drawable.al_imageless_rich_message_price_border));
+            }
+
+            if (!TextUtils.isEmpty(payload.getOverlayText())) {
+                viewHolder.productPrice.setText(payload.getOverlayText());
+                viewHolder.productPrice.setVisibility(View.VISIBLE);
+            } else {
+                viewHolder.productPrice.setVisibility(View.GONE);
+            }
+
+            if (TextUtils.isEmpty(payload.getRating())) {
+                viewHolder.productNameSplitLayout.setVisibility(GONE);
+                if (!TextUtils.isEmpty(payload.getTitle())) {
+                    viewHolder.productNameSingleLine.setVisibility(View.VISIBLE);
+                    viewHolder.productNameSingleLine.setText(payload.getTitle());
+                } else {
+                    viewHolder.productNameSingleLine.setVisibility(View.GONE);
+                }
+            } else {
+                viewHolder.productNameSplitLayout.setVisibility(View.VISIBLE);
+                viewHolder.productNameSingleLine.setVisibility(View.GONE);
+                viewHolder.productName.setVisibility(View.VISIBLE);
+                viewHolder.productRating.setText(payload.getRating());
+
+                if (!TextUtils.isEmpty(payload.getTitle())) {
+                    viewHolder.productName.setText(payload.getTitle());
+                } else {
+                    viewHolder.productName.setText("");
+                }
+            }
+
+            if (!TextUtils.isEmpty(payload.getSubtitle())) {
+                viewHolder.productLocation.setVisibility(View.VISIBLE);
+                viewHolder.productLocation.setText(AlRichMessage.getHtmlText(payload.getSubtitle()));
+            } else {
+                viewHolder.productLocation.setVisibility(View.GONE);
+            }
+
+            if (!TextUtils.isEmpty(payload.getDescription())) {
+                viewHolder.productDescription.setVisibility(View.VISIBLE);
+                viewHolder.productDescription.setText(AlRichMessage.getHtmlText(payload.getDescription()));
+            } else {
+                viewHolder.productDescription.setVisibility(View.GONE);
+            }
+
+            viewHolder.bookAction1.setVisibility(View.GONE);
+            viewHolder.bookAction2.setVisibility(View.GONE);
+            viewHolder.bookAction3.setVisibility(View.GONE);
+            viewHolder.viewAction1.setVisibility(View.GONE);
+            viewHolder.viewAction2.setVisibility(View.GONE);
+            viewHolder.viewAction3.setVisibility(View.GONE);
+
+            if (payload.getActions() != null && !payload.getActions().isEmpty()) {
+                try {
+                    List<ALRichMessageModel.AlButtonModel> actionsList = payload.getActions();
+                    for (int i = 0; i < actionsList.size(); i++) {
+                        if (i == 0) {
```

Comment:
improve code so that we do not need multiple if i == N conditions

Reply:
improved